### PR TITLE
Fixed tests output leak

### PR DESCRIPTION
**Pull request by bocajnotnef**
see https://github.com/dib-lab/khmer/issues/1125

### CODE REVIEW

**Comment by bocajnotnef** [comment]
@ctb @luizirber CR/merge, please?

**Comment by bocajnotnef** [comment]
also https://github.com/dib-lab/khmer/issues/1129

File: tests/test_functions.py [hunk 1]
```diff
@@ -153,19 +154,30 @@ def test_extract_hashbits_info():
...
+    old_stderr = sys.stderr
+    sys.stderr = capture = StringIO()
+    
     try:
         check_input_files(fn, False)
     except SystemExit:
-        check_file_status_exited = True
-    assert check_file_status_exited
-
+        sys.stderr = old_stderr
```
Comment (ctb) [inline tests/test_functions.py]: I would suggest using try/finally here:

```
try:
   ...
except:
   ...
finally:
   sys.stderr = old_stderr

File: tests/test_functions.py [hunk 2]
```diff
@@ -153,19 +154,30 @@ def test_extract_hashbits_info():
...
     fn = utils.get_temp_filename('thisfiledoesnotexist')
+    
+    old_stderr = sys.stderr
+    sys.stderr = capture = StringIO()
+
     try:
         check_input_files(fn, True)
     except OSError:
         assert False
+
```
Comment (ctb) [inline tests/test_functions.py]: and use `finally` here, too.

File: tests/test_normalize_by_median.py [hunk 3]
```diff
@@ -581,7 +580,8 @@ def test_diginorm_basic_functionality_4():
                     infile)
 
     script = 'normalize-by-median.py'
-    args = list(CUTOFF) + list(PAIRING) + ['-k', '15', infile]
+    args = list(CUTOFF) + ['-k', '15', infile]
```
Comment (ctb) [inline tests/test_normalize_by_median.py]: why this change?

File: khmer/khmer_args.py [hunk 4]
```diff
@@ -151,8 +151,7 @@ def create_nodegraph(args, ksize=None, multiplier=1.0):
     if ksize is None:
         ksize = args.ksize
     if ksize > 32:
-        print_error("\n** ERROR: khmer only supports k-mer sizes <= 32.\n")
-        sys.exit(1)
+        raise Exception("** ERROR: khmer only supports k-mer sizes <= 32.")
```
Comment (ctb) [inline khmer/khmer_args.py]: ok, but I *wanted* it to be a print_error, not an Exception...

Comment (bocajnotnef) [inline khmer/khmer_args.py]: Ahh, sorry. I'll switch it back to the old behavior and use pipe redirection to catch the errs.

File: tests/test_normalize_by_median.py (hunk 3 above)
Comment (bocajnotnef) [inline tests/test_normalize_by_median.py]: requiring pairing error'd out--there's unpaired reads in the input (which makes sense since -p used to be used to include single sets of paired/unpaired stuff and this test was never actually running, so it didn't break when we made the change within NBM)

**Comment by bocajnotnef** [comment]
@ctb Updated

**Comment by bocajnotnef** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by bocajnotnef** [comment]
Weird, this shouldn't be mergable as it is...

File: tests/test_normalize_by_median.py (hunk 3 above)
Comment (ctb) [inline tests/test_normalize_by_median.py]: ok thx, good catch :)

File: ChangeLog [hunk 5]
```diff
@@ -1,3 +1,10 @@
+2015-06-30  Jacob Fenton  <bocajnotnef@gmail.com>
+
+   * khmer/khmer_args.py: Changed some error printing to be exceptions
```
Comment (ctb) [inline ChangeLog]: Fix ChangeLog to reflect new commits, pls!

**Comment by ctb** [comment]
Apart from the ChangeLog fix, this LGTM.  Please go ahead and merge after updating the changelog.

**Comment by ctb** [comment]
And thanks for the pep8 fixes...!

**Comment by bocajnotnef** [comment]
Merged #1137
